instrumentation/methods/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/methods/MethodInstrumentationModule.java

```diff
@@ -36,11 +31,10 @@ public MethodInstrumentationModule() {
   }
 
   private static List<TypeInstrumentation> createInstrumentations() {
-    InstrumentationConfig config = AgentInstrumentationConfig.get();
     List<TypeInstrumentation> list =
-        config.isDeclarative()
-            ? MethodsConfig.parseDeclarativeConfig(config.getDeclarativeConfig("methods"))
-            : parseConfigProperties();
+        MethodsConfig.parse(
+            DeclarativeConfigUtil.getStructured(GlobalOpenTelemetry.get(), "java", empty())
+                .getStructured("methods", empty()));
```

Comment:
this was a weird case and had to be converted in this initial PR, because now `config.isDeclarative()` is always true

Reply:
I ended up making several changes in this PR compared to what was in the full PR (see DeclarativeConfigUtil in particular). At this point I'm just treating the full PR as a roadmap and proof that things can work in the end, but I'll be fine-tuning things as I go through the smaller PRs.

Reply:
`config.isDeclarative()` is being removed in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/15659